app/models/journeyDomain/routeDetails/locationOfGoods/LocationOfGoodsDomain.scala

```diff
@@ -73,7 +73,8 @@ object LocationOfGoodsDomain {
 
   case class LocationOfGoodsX(
     typeOfLocation: LocationType,
-    identificationNumber: String
+    identificationNumber: String,
+    addIdentifier: Boolean
```

Comment:
Rather than the boolean this will actually be the identifier itself which will be of type Option[String]. If it's None then we know they've selected No to the addIdentifier question and conversely if it's Some we know they answered Yes.